File: Courses/Bedu/DesarrolloMobileJava/Sesion6/Sesion6/app/src/main/java/xyz/eliothmonroy/sesion6/recyclerview/HomeActivity.java
package xyz.eliothmonroy.sesion6.recyclerview;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import java.util.ArrayList;

import xyz.eliothmonroy.sesion6.R;

public class HomeActivity extends AppCompatActivity {

    ArrayList<String> lista;
    RecyclerView myRecycler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        myRecycler=findViewById(R.id.recycler);
        myRecycler.setLayoutManager(new GridLayoutManager(this,2));
        lista=new ArrayList<>();
        for(int i=0;i<50;i++){
            lista.add("Item num: "+(i+1));
        }
        myRecycler.setAdapter(new MyAdapter(lista));
    }
}

File: Bedu/KotlinFundamentals/Sesion6/retrofitDemo/app/src/main/java/xyz/eliothmonroy/retrofitdemo/reto1/HomeActivity.kt
package xyz.eliothmonroy.retrofitdemo.reto1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import com.google.gson.Gson
import kotlinx.android.synthetic.main.activity_home.*
import org.json.JSONObject
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import xyz.eliothmonroy.retrofitdemo.R
import xyz.eliothmonroy.retrofitdemo.reto1.api.API
import xyz.eliothmonroy.retrofitdemo.reto1.api.RetrofitUtils

class HomeActivity : AppCompatActivity() {

    private val retrofit=RetrofitUtils.getRetofitInstance()
    private lateinit var apiWiki:API

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)

        apiWiki=retrofit.create(API::class.java)

        buscar.setOnClickListener {
            if(busqueda.text.toString()!=""){
                obtenerResultadosApi(busqueda.text.toString())
            }else{
                Toast.makeText(this@HomeActivity,"Por favor ingrese algún término", Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun obtenerResultadosApi(value:String) {
        val resultado=apiWiki.getValueApiWiki("query","extracts","json","",value)
        resultado.enqueue(object : Callback<Any> {
                override fun onFailure(call: Call<Any>, t: Throwable) {
                    Toast.makeText(this@HomeActivity,"Error: $t", Toast.LENGTH_SHORT).show()
                }
                override fun onResponse(
                    call: Call<Any>,
                    response: Response<Any>
                ) {
                    if (response.isSuccessful){
                        response.body()?.let {
                            val objeto=JSONObject(Gson().toJson(response.body()))
                            val query=JSONObject(objeto["query"].toString())
                            val pages=JSONObject(query["pages"].toString())
                            val clave=pages.keys()
                            val wrapper=JSONObject(pages[clave.next()].toString())
                            Log.d("Home",wrapper.toString())
                            webViewWiki.loadData(wrapper["extract"].toString(),"text/html","UTF-8")
                        }
                    }
                }
            }
        )

    }
}
